pkg/virt-handler/virtwrap/manager.go

```diff
@@ -328,21 +327,14 @@ func VMNamespaceKeyFunc(vm *v1.VM) string {
 	return domName
 }
 
-func (l *LibvirtDomainManager) SyncVM(vm *v1.VM) (*api.DomainSpec, error) {
-	var wantedSpec api.DomainSpec
-	mappingErrs := model.Copy(&wantedSpec, vm.Spec.Domain)
-
-	if len(mappingErrs) > 0 {
-		return nil, errors.NewAggregate(mappingErrs)
-	}
-
+func (l *LibvirtDomainManager) SyncVM(vm *v1.VM, domDesign *designer.DomainDesign) (*api.DomainSpec, error) {
```

Comment:
As I said, it might be necessary to revisit some of this (which does not mean that it can't work this way), when we allow more different network types. Right now, it is not necessary. There are different solutions how to represent networks in the cluster. We would still have the network device already on the VM spec, where we want to connect the network.

Please provide concrete examples. As said, I am very interested in this and will be happy to go through each one.